include/dpp/cache.h

```diff
@@ -269,5 +264,53 @@ cache_decl(role, find_role, get_role_cache, get_role_count);
 cache_decl(channel, find_channel, get_channel_cache, get_channel_count);
 cache_decl(emoji, find_emoji, get_emoji_cache, get_emoji_count);
 
+/** Run garbage collection across all caches removing deleted items
+ * that have been deleted over 60 seconds ago.
+ */
+ /* Because other threads and systems may run for a short while after an event is received, we don't immediately
+  * delete pointers when objects are replaced. We put them into a queue, and periodically delete pointers in the
+  * queue. This also rehashes unordered_maps to ensure they free their memory.
+  */
+  /* Because other threads and systems may run for a short while after an event is received, we don't immediately
+   * delete pointers when objects are replaced. We put them into a queue, and periodically delete pointers in the
+   * queue. This also rehashes unordered_maps to ensure they free their memory.
+   */
+template<typename value_type> inline void garbage_collection() {
+	time_t now = time(nullptr);
```

Comment:
If we're really crunching for memory, would something like:
```c++
time_t now;
time(&now);
```
not be better? [See godbolt here](https://godbolt.org/z/E9KEhETx9).

I'm still pretty new to all of this so I could be wrong!

Reply:
> I'm not sure - why?

Less instructions, less of the `QWORD PTR [rbp-8]`, I'm not well versed in assembly (actually, never done assembly lol) but from a bit of googling, the suggestion seems to be better but only if you're really optimising (so modern architecture will probably not see this change performance whatsoever).